Functions/Mock.ps1

```diff
@@ -232,8 +232,9 @@ about_Mocking
                 $cmdletBinding = $cmdletBinding.Insert($cmdletBinding.Length - 2, 'PositionalBinding=$false')
             }
 
-            $repairedMetadata = Repair-ConflictingParameters -Metadata $metadata
-            $paramBlock = [Management.Automation.ProxyCommand]::GetParamBlock($repairedMetadata)
+            # Will modify $metadata object in-place
+            Repair-ConflictingParameters -Metadata $metadata
```

Comment:
This is a cosmetic change from the #1007 implementation. We are now making explicit the fact that the metadata object is modified in place